pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -717,7 +689,7 @@ def _save_none_monitor_checkpoint(self, trainer: 'pl.Trainer', monitor_candidate
         self._save_model(trainer, filepath)
 
         if (
-            self.save_top_k is None and self.best_model_path and self.best_model_path != filepath
+            self.save_top_k == 1 and self.best_model_path and self.best_model_path != filepath
```

Comment:
Are you sure there ?

Reply:
Yes. What's not clear?

`save_top_k==0`: there's an early exit above
`save_top_k==-1`: we don't delete checkpoints
`save_top_k>1`: Not a valid configuration with `monitor==None`. Misconfig would have been raised.